src/BlazorWebView/src/Maui/Android/AndroidWebKitWebViewManager.cs

```diff
@@ -66,7 +66,13 @@ internal void SetUpMessageChannel()
 
 			var nativeToJs = new BlazorWebMessageCallback(message =>
 			{
-				MessageReceived(new Uri(AppOrigin), message!);
+				var currentUri = _webview.Url != null ? new Uri(_webview.Url) : null;
+				var appOriginUri = new Uri(AppOrigin);
```

Comment:
Unrelated to this PR: Would it make sense to also use a static for `new Uri(AppOrigin)` similar to `AndroidAppOriginUri`?